service/src/main/java/uk/nhs/adaptors/gpc/consumer/filters/RoutingGatewayFilterFactory.java

```diff
@@ -0,0 +1,75 @@
+package uk.nhs.adaptors.gpc.consumer.filters;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.OrderedGatewayFilter;
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
+import org.springframework.cloud.gateway.route.RouteLocator;
+import org.springframework.cloud.gateway.route.builder.RouteLocatorBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+import uk.nhs.adaptors.gpc.consumer.utils.FindAPatientDocsUtil;
+
+@Configuration
+@Slf4j
+public class RoutingGatewayFilterFactory extends AbstractGatewayFilterFactory<RoutingGatewayFilterFactory.Config> {
+    private static final int PRIORITY = -3;
+    @Value("${gpc-consumer.gpc.gpcConsumerUrl}")
+    private String gpcConsumerUrl;
+    @Value("${gpc-consumer.gpc.gpcUrl}")
+    private String gpcUrl;
+    @Value("${gpc-consumer.gpc.searchForAPatientsDocumentsPath}")
+    private String searchForAPatientsDocumentsPath;
+    @Value("${gpc-consumer.gpc.structuredPath}")
+    private String structuredPath;
+    @Value("${gpc-consumer.gpc.findPatientPath}")
+    private String findPatientPath;
+    @Value("${gpc-consumer.gpc.documentPath}")
+    private String documentPath;
+
+    @Override
+    public GatewayFilter apply(Config config) {
+        return new OrderedGatewayFilter((exchange, chain) ->
+            chain.filter(exchange.mutate().response(exchange.getResponse()).build()), PRIORITY);
+    }
+
+    @Bean
+    public RouteLocator routes(RouteLocatorBuilder builder) {
+        return builder.routes()
+            .route("get-document", r -> r.path(documentPath)
+                .and()
+                .uri(gpcUrl + documentPath))
```

Comment:
Are we sure this is the correct usage of uri()? In application.yml it was just the http://hostname:port/. This tutorial says the same. https://medium.com/@niral22/spring-cloud-gateway-tutorial-5311ddd59816 Official docs are lacking here though.

Reply:
Getting started guide uses java config and doesn't put the context path in uri: https://spring.io/guides/gs/gateway/

Reply:
I struggled with docs too for this, I debugged it and it actually only carried across the http://hostname:port/
This is now updated